### ISSUE TITLE
How to unroot Waydroid?

### ISSUE THREAD

**Issue by puspamadak** [waydroid/waydroid#935]
**Describe the bug**
Waydroid is a rooted system, which means all the system-level files are accessible from apps. while this is a good thing for some people, many apps have root checker which detects the device is rooted & doesn't let use it.  
A screenshot is provided of one such app. Is there any way to disable root access entirely or for specific apps? I have already tried following several blog posts, but the instructions are given for Magisk, so they do not work in Waydroid.

**Screenshots**
![image](https://github.com/waydroid/waydroid/assets/59383614/e24f18ad-69b1-456f-960e-9be03a0c4187)

**General information (please complete the following information):**
 - Waydroid tools Version: 1.4.1
 - Waydroid Images Version `system_datetime`/`vendor_datetime`: 0/0

**Desktop (please complete the following information):**
 - OS: Arch
 - GPU: Intel HD Graphics 2500
 - Kernel version: Linux Arch 6.1.31-1-lts
 - Host mesa version: 23.1.1-1
 - Desktop: Gnome 44.1

**Smartphone (please complete the following information):**
 - Device: WayDroid x86_64 Device
 - OS: LineageOS
 - Mainline or Halium: Mainline
 - Kernel version: 6.1.31-1-lts

**Comment by Quackdoc** [comment]
waydroid is not rooted your app is most likely looking at waydroid is an emulator, report it to the application

**Comment by vishvaamitra** [comment]
Is there anyway to mask this from the app?

**Comment by puspamadak** [comment]
> waydroid is not rooted your app is most likely looking at waydroid is an emulator, report it to the application

IMO waydroid or it's image is certainly rooted, because the system files are readable which are usually shown as permission denied in non-rooted devices.

**Comment by stangerm2** [comment]
Same, can't use company outlook:
file:///home/matt/Pictures/Screenshots/Screenshot%20from%202023-08-11%2010-36-00.png
I've even tried https://github.com/nitanmarcel/waydroid-magisk but still it's detected as rooted.

**Comment by viibhuGupta** [comment]
i also get this Problem do you get the solution Its work well in Android Studio Emaulator

**Comment by g1i7ch** [comment]
i would also love a solution to this

**Comment by viibhuGupta** [comment]
> i would also love a solution to this

Did you get the solution please reply

**Comment by Quackdoc** [comment]
can you try https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955

**Comment by puspamadak** [comment]
Thanks a lot @Quackdoc it worked!

**Comment by baptx** [comment]
@Quackdoc the change you mentioned is working to get rid of the custom ROM detection popup of WhatsApp / WhatsApp Business but it is not enough for other Android apps using [rootbeer](https://github.com/scottyab/rootbeer) for root detection.
We can see that Waydroid is detected as a rooted device with https://github.com/KimChangYoun/rootbeerFresh even after editing waydroid_base.prop because it still fails the steps "dangerous props" and "SE Linux flag is enabled".
Any idea how to bypass that?

**Comment by YogSottot** [comment]
> dangerous props
```

    public boolean checkForDangerousProps() {

        final Map<String, String> dangerousProps = new HashMap<>();
        dangerousProps.put("ro.debuggable", "1");
        dangerousProps.put("ro.secure", "0");
```

check https://github.com/waydroid/waydroid/pull/1096/files

> SE Linux flag is enabled"

It is broken. See issue 11 in the rootbeerFresh repository

```
public static boolean isSelinuxFlagInEnabled() {
        try {
            Class<?> c = Class.forName("android.os.SystemProperties");
            Method get = c.getMethod("get", String.class);
            String selinux = (String) get.invoke(c, "ro.build.selinux");
            return "1".equals(selinux);
        } catch (Exception ignored) {
```

**Comment by baptx** [comment]
@YogSottot  Thanks, I thought "dangerous props" meant the dangerous props were already used in `/var/lib/waydroid/waydroid_base.prop` but I could not find them. Adding these lines to waydroid_base.prop bypassed root detection of rootbeerFresh:
```
ro.adb.secure=1
ro.debuggable=0
ro.build.selinux=1
```
However it looks like there are still issues to pass SafetyNet (and its replacement Play Integrity):
https://github.com/waydroid/waydroid/issues/182
I installed Play Integrity API Checker from Google Play Store and it did not pass the tests with the modified waydroid_base.prop.
Even LineageOS used by Waydroid does not pass SafetyNet:
https://www.lineageos.org/Safetynet/
It would be good to know how SafetyNet / Play Integrity detects the custom ROM so we can try to pass the test.

**Comment by Thesola10** [comment]
Apple Music wouldn't launch anymore after a Waydroid update, these props get RootBeerFresh and Apple Music to see Waydroid as non-rooted:
```
ro.adb.secure=1
ro.debuggable=0
ro.build.selinux=1
ro.build.tags=release-keys
ro.product.build.tags=release-keys
ro.vendor.build.tags=release-keys
ro.odm.build.tags=release-keys
```

Adding them to `/var/lib/waydroid/waydroid_base.prop` should be enough. Also run `chmod g-w o-w -R /var/lib/waydroid/overlay` for good measure if you installed an overlay such as Widevine or Houdini.

**Comment by LFRon** [comment]
In ro.build.selinux=1

Could this force open the SELinux on waydroid?
I remembered images at 2022.12 could force use SELinux inside waydroid though my Ubuntu uses AppArmor😂

**Comment by kode54** [comment]
The flag is a lie in this case.

**Comment by RokeJulianLockhart** [comment]
Would this be worth closing as a duplicate of https://github.com/waydroid/waydroid/issues/1231#issue-2071631747?

**Comment by kode54** [comment]
This comment looks easier to read: https://github.com/waydroid/waydroid/issues/935#issuecomment-1961262639 for faking root.

**Comment by Fuseteam** [comment]
> Could this force open the SELinux on waydroid? I remembered images at 2022.12 could force use SELinux inside waydroid 

@LFRon Was this on fedora? i recall someone was trying that in #687 but didn't succeed

**Comment by EyBmr** [comment]
Hey! Anybody got things working? I have tried all of everyone's suggestions but nothing worked.

My "comdirect photoTAN" online banking app is still detecting root and exits immediately.

**Comment by Fuseteam** [comment]
maybe try this: https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955

**Comment by EyBmr** [comment]
> maybe try this: [#1060 (comment)](https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955)

Nope. Doesn't work at all. My baking app still detects root and immediately quits. Using the latest LineageOS GAPPS image (lineage-18.1-20241005-GAPPS).

**Comment by EyBmr** [comment]
> maybe try this: [#1060 (comment)](https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955)

Nope. Doesn't work at all. My banking app still detects root and immediately quits. Using the latest LineageOS GAPPS image (lineage-18.1-20241005-GAPPS).

**Comment by EyBmr** [comment]
Nope. Doesn't work at all. My banking app still detects root and immediately quits and I can't do anything about it. Using the latest LineageOS GAPPS image (lineage-18.1-20241005-GAPPS).

**Comment by EyBmr** [comment]
> maybe try this: [#1060 (comment)](https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955)

Nope. Doesn't work at all. My banking app still detects root and immediately quits and I can't do anything about it. Using the latest LineageOS GAPPS image (lineage-18.1-20241005-GAPPS).

**Comment by EyBmr** [comment]
> maybe try this: [#1060 (comment)](https://github.com/waydroid/waydroid/issues/1060#issuecomment-1732325955)

Nope. Doesn't work at all. My banking app still detects root and immediately closes itself and I can't do anything about it. Using the latest LineageOS GAPPS image (lineage-18.1-20241005-GAPPS).

**Comment by exu-g** [comment]
Is there any reason to not ship these prop workarounds by default?  
I'd say it would be an improvement in most cases and reduce the number of these issues
